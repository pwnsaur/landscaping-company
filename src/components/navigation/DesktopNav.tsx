import styled from 'styled-components';
import NavItems from '@/components/navigation/NavItems';

const DesktopNav = () => {
  return (
    <StyledDesktopNav>
      <NavItems />
    </StyledDesktopNav>
  );
};

export default DesktopNav;

const StyledDesktopNav = styled.nav`
  display: flex;
  justify-content: space-around;
  width: 100%;
  height: 100%;
  max-width: 48rem;
`;
