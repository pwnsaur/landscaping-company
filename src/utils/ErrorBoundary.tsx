import React, { ErrorInfo } from 'react';

interface Props {
  children: React.ReactNode;
}

interface State {
  hasError: boolean;
}

class ErrorBoundary extends React.Component<Props, State> {
  constructor(props: Props) {
    super(props);

    this.state = { hasError: false };
  }

  static getDerivedStateFromError(_: Error): State {
    return { hasError: true };
  }

  componentDidCatch(error: Error, errorInfo: ErrorInfo) {
    console.log({ error, errorInfo });
  }

  render() {
    if (this.state.hasError) {
      return (
        <div>
          <h1>Lapu neizdevās ielādēt.</h1>
        </div>
      );
    }

    return this.props.children;
  }
}

export default ErrorBoundary;
