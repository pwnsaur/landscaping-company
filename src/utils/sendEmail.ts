import axios, { AxiosResponse } from 'axios';

import { FormData } from '@/types/contactForm';

type ServerResponse = {
  messageId: string;
  response: string;
};

type SendEmailFn = (
  formData: FormData
) => Promise<AxiosResponse<ServerResponse>>;

const sendEmail: SendEmailFn = async (formData: FormData) => {
  return axios({
    method: 'post',
    url: '/api/send-mail',
    data: formData,
  });
};

export default sendEmail;
