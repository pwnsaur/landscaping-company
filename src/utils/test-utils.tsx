import { ReactElement, ReactNode } from 'react';
import { ThemeProvider } from 'styled-components';

import { theme } from '@/styles/theme';
import useIsMobile from '@/utils/hooks/useIsMobile';
import { render, RenderOptions } from '@testing-library/react';

interface TestWrapperProps {
  children: ReactNode;
}

const TestWrapper = ({ children }: TestWrapperProps) => {
  const isMobile = useIsMobile();

  return (
    <ThemeProvider theme={{ ...theme, isMobile }}>{children}</ThemeProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'queries'>
) => {
  return render(ui, { wrapper: TestWrapper, ...options });
};

// eslint-disable-next-line import/export
export * from '@testing-library/react';

// eslint-disable-next-line import/export
export { customRender as render };
