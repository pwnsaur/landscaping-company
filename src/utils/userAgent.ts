export function isMobileUserAgent(userAgent: string): boolean {
  return /Android|iPhone|iPod|BlackBerry|IEMobile|Opera Mini/i.test(userAgent);
}
