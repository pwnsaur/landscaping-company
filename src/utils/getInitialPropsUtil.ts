import { AppContext, AppInitialProps } from 'next/app';

import { isMobileUserAgent } from './userAgent';

export async function getInitialPropsUtil(
  appContext: AppContext
): Promise<AppInitialProps & { isMobile: boolean }> {
  let pageProps = {};

  if (appContext.Component.getInitialProps) {
    pageProps = await appContext.Component.getInitialProps(appContext.ctx);
  }

  const userAgent = appContext.ctx.req?.headers['user-agent'] || 'rekt';
  const isMobile = isMobileUserAgent(userAgent);

  return { pageProps, isMobile };
}
