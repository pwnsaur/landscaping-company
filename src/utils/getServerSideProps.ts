import { GetServerSideProps } from 'next';

import { isMobileUserAgent } from '@utils/userAgent';

export const getServerSideProps: GetServerSideProps = async ({ req }) => {
  const userAgent = req.headers['user-agent'] || 'about rekt';
  const isMobile = isMobileUserAgent(userAgent);
  return { props: { isMobile } };
};
