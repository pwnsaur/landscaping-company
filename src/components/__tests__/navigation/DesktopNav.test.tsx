import { render } from '@/utils/test-utils';
import DesktopNav from '@/components/navigation/DesktopNav';

describe('MobileNav component', () => {
  test('matches the snapshot', () => {
    const { asFragment } = render(<DesktopNav />);
    expect(asFragment()).toMatchSnapshot();
  });
});
