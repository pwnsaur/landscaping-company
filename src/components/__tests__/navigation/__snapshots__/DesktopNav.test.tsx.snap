// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileNav component matches the snapshot 1`] = `
<DocumentFragment>
  <nav
    class="sc-23f60e78-0 gVZxNu"
  >
    <a
      class="sc-a1402bc8-0 NAwpH"
      href="/"
    >
      SƒÅkums
    </a>
    <a
      class="sc-a1402bc8-0 NAwpH"
      href="/services"
    >
      Pakalpojumi
    </a>
    <a
      class="sc-a1402bc8-0 NAwpH"
      href="/projects"
    >
      Projekti
    </a>
    <a
      class="sc-a1402bc8-0 NAwpH"
      href="/about"
    >
      Par mums
    </a>
    <a
      class="sc-a1402bc8-0 NAwpH"
      href="/contacts"
    >
      Kontakti
    </a>
  </nav>
</DocumentFragment>
`;
